SkillsFundingAgency/das-apprentice-commitments-api
# Separate (approvals) registration from (apprentice) account

This PR introduces a separation between creating an account for an apprentice, and matching the apprentice to an apprenticeship registration (approval).

There are now these main routes:

- Register the apprenticeship approval

      POST /registrations
      {
          "registrationId": "{{registrationId}}",
          "commitmentsApprenticeshipId": {{approvalsId}},
          "commitmentsApprovedOn": "{{approvedOn}}",
          "firstName": "{{firstName}}",
          "lastName": "{{lastName}}",
          "dateOfBirth": "{{dob}}",
          "email": "{{email}}",
          "employerName": "TestEmployerCo",
          "employerAccountLegalEntityId": 124,
          "trainingProviderId": "222",
          "trainingProviderName": "TestProviderAc",
          "courseName": "Learningy",
          "courseLevel": 2,
          "courseDuration": 24,
          "courseOption": null,
          "plannedStartDate": "2022-08-01T00:00:00",
          "plannedEndDate": "0001-01-01T00:00:00"
      }

- Create the apprentice account

      POST /apprentices
      {
        "apprenticeId": "{{apprenticeId}}",
        "firstName": "{{firstName}}",
        "lastName": "{{lastName}}",
        "dateOfBirth": "{{dob}}",
        "email": "{{email}}"
      }

- Match the apprentice to the apprenticeship

      POST /apprenticeships
      {
          "registrationId": "{{registrationId}}",
          "apprenticeId": "{{apprenticeId}}"
      }

- Confirm the apprenticeship

      PATCH /apprentices/{{apprenticeId}}/apprenticeships/{{apprenticeshipId}}/revisions/{{revisionId}}/confirmations
      {
          "employerCorrect": "true",
          "providerCorrect": "true",
          "apprenticeshipDetailsCorrect": "true",
          "deliveryConfirmed": "true",
          "rolesAndResponsibilitiesConfirmed": "true"
      }
